core/src/mast/node/mod.rs

```diff
@@ -236,6 +236,20 @@ impl MastNode {
 
 /// Mutators
 impl MastNode {
+    /// Sets the list of decorators for the [`MastNode`]. Currently, is only supported for the
+    /// [`Block`] variant.
+    pub fn set_decorators(&mut self, decorator_list: DecoratorList) {
```

Comment:
As mentioned, I would remove this in favor of a `match` at the call-site so we can avoid adding this public method and we can handle errors in a better way.